internal/servers/worker/session.go

```diff
@@ -152,6 +152,28 @@ func (w *Worker) activateSession(ctx context.Context, sessionId, tofuToken strin
 	return resp.GetStatus(), nil
 }
 
+func (w *Worker) cancelSession(ctx context.Context, sessionId string) (pbs.SESSIONSTATUS, error) {
+	rawConn := w.controllerSessionConn.Load()
+	if rawConn == nil {
+		return pbs.SESSIONSTATUS_SESSIONSTATUS_UNSPECIFIED, errors.New("could not get a controller client")
+	}
+	conn, ok := rawConn.(pbs.SessionServiceClient)
+	if !ok {
+		return pbs.SESSIONSTATUS_SESSIONSTATUS_UNSPECIFIED, errors.New("could not cast atomic controller client to the real thing")
+	}
+	if conn == nil {
```

Comment:
Can this ever be nil, given the above two checks?

Reply:
Yep! One of those tricksy Go things. The value back from the atomic.Value is an `interface{}` which can be nil, that is, `interface{}(nil)`. But it can also be a non-empty reference to a different interface variable, but where that interface variable itself is `nil`, that is, `interface{}(pbs.SessionServiceClient(nil))`.